### PR TITLE
Feature/jjdd8fpo 26a

### PR DESCRIPTION
**Pull request by lukaszwolek**
wczytywanie danych z pliku

### CODE REVIEW

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: web/src/main/java/com/infoshareacademy/domain/api/AuthorJson.java [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 public class AuthorJson {
 
+    @JsonProperty("author")
     private String name;
 
+    //private String name;
+
```
Comment (DariuszCzajkowski) [inline web/src/main/java/com/infoshareacademy/domain/api/AuthorJson.java]: usuń komentarz

File: web/src/main/java/com/infoshareacademy/service/AuthorService.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        Author author = authorDao.findAuthorByName(name);
+        if (author == null) {
+            Author newAuthor = new Author();
+            newAuthor.setName(name);
+            authorDao.addAuthor(newAuthor);
+            System.out.println("New author added: " + newAuthor.getName());
+            return newAuthor;
+        }
+        System.out.println("Author added: " + author.getName());
+        return author;
```
Comment (DariuszCzajkowski) [inline web/src/main/java/com/infoshareacademy/service/AuthorService.java]: Sout -ów nie używamy

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: web/src/main/java/com/infoshareacademy/service/EpochService.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public class EpochService {
+
+    @Inject
+    private EpochDao epochDao;
+
+    public Epoch findOrAdd(String name)  {
+
+        Epoch epoch = epochDao.findEpochByName(name);
+        if (epoch == null)  {
+            Epoch newEpoch = new Epoch();
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshareacademy/service/EpochService.java]: po co tworzysz nowy obiekt z nową referencją? z powodzeniem użyjesz epoch. Wtedy tylko jeden return CI wystarczy

File: web/src/main/java/com/infoshareacademy/service/GenreService.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+public class GenreService {
+
+    @Inject
+    private GenreDao genreDao;
+
+    public Genre findOrAdd(String name) {
+
+        Genre genre = genreDao.findGenreByName(name);
+        if (genre == null) {
+            Genre newGenre = new Genre();
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshareacademy/service/GenreService.java]: jw

File: web/src/main/java/com/infoshareacademy/service/KindService.java [hunk 5]
```diff
@@ -14,15 +14,26 @@
...
     public void addKind(String kindName) {
         LiteratureKind k = new LiteratureKind();
+
         k.setName(kindName);
         kindDao.addKind(k);
+    }
+
+    public LiteratureKind findOrAdd(String name)  {
 
+        LiteratureKind kind = kindDao.findKindByName(name);
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshareacademy/service/KindService.java]: jw

File: web/src/main/java/com/infoshareacademy/servlet/LoaderFromFileServlet.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        Part file = req.getPart("api");
+        List<BookDTO> books = new ArrayList<>();
+        try {
+            books = parserService.parse(uploaderService.uploadApiFile(file), BookDTO.class);
+        } catch (ApiFileNotFound apiFileNotFound) {
+            apiFileNotFound.printStackTrace();
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshareacademy/servlet/LoaderFromFileServlet.java]: logger....

**Comment by isacademyci** [comment]
Build triggered for original commit.

**Comment by isacademyci** [comment]
Build started for original commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Pull request merged**